src/views/forms/AdjustIsolatedMarginForm.tsx

```diff
@@ -46,46 +58,118 @@ const SIZE_PERCENT_OPTIONS = {
   '75%': '0.75',
 };
 
-export const AdjustIsolatedMarginForm = ({ marketId }: ElementProps) => {
+export const AdjustIsolatedMarginForm = ({
+  marketId,
+  onIsolatedMarginAdjustment,
+}: ElementProps) => {
   const stringGetter = useStringGetter();
-  const [marginAction, setMarginAction] = useState(MarginAction.ADD);
   const subaccountPosition = useAppSelector(getOpenPositionFromId(marketId));
-  const { adjustedMmf, leverage, liquidationPrice, notionalTotal } = subaccountPosition ?? {};
+  const { childSubaccountNumber } = subaccountPosition ?? {};
   const marketConfig = useAppSelector((s) => getMarketConfig(s, marketId));
+  const adjustIsolatedMarginInputs = useAppSelector(getAdjustIsolatedMarginInputs, shallowEqual);
+
+  const {
+    type: isolatedMarginAdjustmentType,
+    amount,
+    amountPercent,
+    summary,
+  } = adjustIsolatedMarginInputs ?? {};
+
   const { tickSizeDecimals } = marketConfig ?? {};
 
-  /**
-   * @todo: Replace with Abacus functionality
-   */
-  const [percent, setPercent] = useState('');
-  const [amount, setAmount] = useState('');
-  const onSubmit = () => {};
+  useEffect(() => {
+    abacusStateManager.setAdjustIsolatedMarginValue({
+      value: childSubaccountNumber,
+      field: AdjustIsolatedMarginInputField.ChildSubaccountNumber,
+    });
 
-  const positionMargin = useMemo(
-    () => ({
-      current: calculatePositionMargin({
-        adjustedMmf: adjustedMmf?.current,
-        notionalTotal: notionalTotal?.current,
-      }).toFixed(tickSizeDecimals ?? USD_DECIMALS),
-      postOrder: calculatePositionMargin({
-        adjustedMmf: adjustedMmf?.postOrder,
-        notionalTotal: notionalTotal?.postOrder,
-      }).toFixed(tickSizeDecimals ?? USD_DECIMALS),
-    }),
-    [adjustedMmf, notionalTotal, tickSizeDecimals]
-  );
+    return () => {
+      abacusStateManager.setAdjustIsolatedMarginValue({
+        value: null,
+        field: AdjustIsolatedMarginInputField.ChildSubaccountNumber,
+      });
+      abacusStateManager.clearAdjustIsolatedMarginInputValues();
+      abacusStateManager.clearTradeInputValues({ shouldResetSize: true });
+    };
+  }, [childSubaccountNumber]);
 
-  const { freeCollateral, marginUsage } = useAppSelector(getSubaccount, shallowEqual) ?? {};
+  const setAmount = ({ floatValue }: NumberFormatValues) => {
+    abacusStateManager.setAdjustIsolatedMarginValue({
+      value: floatValue,
+      field: AdjustIsolatedMarginInputField.Amount,
+    });
+  };
+
+  const setPercent = (value: string) => {
+    abacusStateManager.setAdjustIsolatedMarginValue({
```

Comment:
not blocking -- but currently i can select a percent option but it doesn't look "selected"